File: src/main/java/com/axonivy/market/model/ProductDetailModel.java
package com.axonivy.market.model;

import com.axonivy.market.entity.ReadmeProductContent;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@NoArgsConstructor
public class ProductDetailModel extends ProductModel {
    private String vendor;
    private String vendorUrl;
    private String platformReview;
    private String newestReleaseVersion;
    private String cost;
    private String sourceUrl;
    private String statusBadgeUrl;
    private String language;
    private String industry;
    private String compatibility;
    private Boolean contactUs;
    private List<ReadmeProductContent> readmeProductContents;
}

File: marketplace-ui/src/app/shared/models/product-detail.model.ts
import { DisplayValue } from './display-value.model';
import { ProductModuleContent } from './product-module-content.model';

export interface ProductDetail {
  id: string;
  names: DisplayValue;
  shortDescriptions: DisplayValue;
  logoUrl: string;
  type: string;
  tags: string[];
  vendor: string;
  vendorUrl: string;
  vendorImage: string;
  vendorImageDarkMode: string;
  platformReview: string;
  newestReleaseVersion: string;
  cost: string;
  sourceUrl: string;
  statusBadgeUrl: string;
  language: string;
  industry: string;
  contactUs: boolean;
  installationCount: number;
  productModuleContent: ProductModuleContent;
  mavenDropins: boolean;
  metaProductJsonUrl?: string;
  compatibilityRange?: string;
  isFocusedProduct: boolean;
  _links: {
    self: {
      href: string;
    };
  };
}
